test/unit/app/services/bulk_republisher_test.rb

```diff
@@ -201,4 +201,40 @@ class BulkRepublisherTest < ActiveSupport::TestCase
       BulkRepublisher.new.republish_all_published_organisation_about_us_pages
     end
   end
+
+  describe "#republish_all_by_type" do
+    setup do
+      BulkRepublisher.any_instance.stubs(:non_editionable_content_types).returns(%w[Contact])
+    end
+
+    context "for editionable content types, like CaseStudy" do
+      test "republishes all content of the specified type via the PublishingApiDocumentRepublishingWorker" do
+        content_type = create(:case_study)
+
+        PublishingApiDocumentRepublishingWorker.expects(:perform_async_in_queue).with(
+          "bulk_republishing",
+          content_type.document_id,
+          true,
+        )
+        BulkRepublisher.new.republish_all_by_type("CaseStudy")
+      end
+
+      context "for non-editionable content types, like Contact" do
+        test "republishes all content of the specified type via the Whitehall::Publishing API" do
+          content_type = create(:contact)
+          Whitehall::PublishingApi.expects(:bulk_republish_async).with(content_type)
+          BulkRepublisher.new.republish_all_by_type("Contact")
+        end
+      end
+
+      describe "for non-existent content types" do
+        test "it returns an error" do
+          content_type = "SomeDocumentTypeThatDoesntExist"
+          assert_raises(StandardError, /Unknown content type #{content_type}/) do
```

Comment:
Per #9149, the string here needs to be in the keyword argument `match`, otherwise it's ignored and any message will pass. It doesn't need to be a `//`-style string either

Reply:
Ah, a useful discovery, nice one!